src/providers/aws.rs

```diff
@@ -1321,6 +1304,31 @@ impl RegionLauncher {
             .instrument(key_span)
             .await;
         }
+
+        if !self.security_group_id.trim().is_empty() {
+            let group_span = tracing::trace_span!("security_group", id = %self.security_group_id);
+            async {
+                tracing::trace!("removing security group.");
+                // clean up security groups and keys
+                loop {
+                    let mut req = rusoto_ec2::DeleteSecurityGroupRequest::default();
+                    req.group_id = Some(self.security_group_id.clone());
+                    if let Err(e) = client.delete_security_group(req).in_current_span().await {
+                        tracing::trace!(
```

Comment:
I think here we want to check _what_ error we got. If we get an error _other_ than "busy" or "in use" or whatever it is, then we should propagate that error. The error match against is:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<Response><Errors><Error><Code>DependencyViolation</Code><Message>resource sg-0c3ddd8f4d8bfe33c has a dependent object</Message></Error></Errors><RequestID>196925f6-b9b9-45ed-b4c6-d07798afca37</RequestID></Response>
```
`DependencyViolation` under `<Code>` seems like what we want to look for.

Reply:
Alternatively, instead of retrying the delete, we can just keep checking the state of all the instances until all of them are in the `terminated` state. And only then try the delete, which should then succeed immediately. That would mean we don't have to match on the error.